### PR TITLE
Refactor CurrentVersions and cache file downloads

### PR DESCRIPTION
**Pull request by sebastienros**
Fixes https://github.com/dotnet/crank/issues/756

### CODE REVIEW

File: src/Microsoft.Crank.Agent/Startup.cs [hunk 1]
```diff
@@ -132,8 +135,9 @@ Based on the target framework
         private static Process perfCollectProcess;
         private static readonly object _synLock = new();
         private static object _consoleLock = new();
+        private static MemoryCache _fileContentCache = new(new MemoryCacheOptions { SizeLimit = 10_000_000 });
 
-        private static Task dotnetTraceTask;
+private static Task dotnetTraceTask;
```
Comment (halter73) [inline src/Microsoft.Crank.Agent/Startup.cs:140]:
```suggestion
        private static Task dotnetTraceTask;
```

File: src/Microsoft.Crank.Agent/Startup.cs [hunk 2]
```diff
@@ -3835,24 +3837,28 @@ private static string SeekCompatibleDesktopRuntime(string dotnetHome, string tar
...
+
                     Log.Info($"ASP.NET: {aspNetCoreVersion} (Current)");
                     break;
                 case "latest":
                     // aspnet runtime service releases are not published on feeds
                     switch (versionPrefix)
                     {
                         case "9.0":
-                            var productsInfo = JObject.Parse(await DownloadContentAsync(_latestProductVersions90Url));
+                            var productsInfo = JObject.Parse(await DownloadContentAsync(_latestProductVersions90Url, cache: _latestProductVersions90Cache));
```
Comment (halter73) [inline src/Microsoft.Crank.Agent/Startup.cs:3861]: It looks like `_latestProductVersions90Cache` is the actual memory cache here. I would rename both the static and the parameter to include timeout.

```suggestion
                            var productsInfo = JObject.Parse(await DownloadContentAsync(_latestProductVersions90Url, cacheTimeout: _latestProductVersions90CacheTimeout));
```

Same goes for `_aspnetSdkVersionCache` -> `_aspnetSdkVersionCacheTimeout`.

**Review by halter73 — APPROVED** [review]

**Comment by joperezr** [comment]
FYI @sebastienros PR was merged due to automerge but @halter73 had some feedback above. Commenting to make sure that isn't missed 😉

**Comment by sebastienros** [comment]
I saw that, thanks

**Pull request merged**